src/item.cpp

```diff
@@ -2535,6 +2535,26 @@ void item::magazine_info( std::vector<iteminfo> &info, const iteminfo_query *par
         return;
     }
 
+    if( parts->test( iteminfo_parts::MAGAZINE_COMPATIBLE_GUNS ) ) {
+        const auto compare = []( const auto & lhs, const auto & rhs ) {
+            return localized_compare( lhs, rhs );
+        };
+        std::set<std::string, decltype( compare )> compatible_guns( compare );
+        for( const itype *it : item_controller->all() ) {
+            const item gun( it->get_id(), calendar::turn );
```

Comment:
This won't work well for gun variants on - this item will randomly select a variant, so the shown guns will randomly swap.
You probably want something like (may need additional comments)
```diff
diff --git a/src/item.cpp b/src/item.cpp
index b0b413b08b..bc9d19109e 100644
--- a/src/item.cpp
+++ b/src/item.cpp
@@ -13597,6 +13597,22 @@ std::string item::type_name( unsigned int quantity ) const
     return ret_name;
 }

+std::vector<std::string> item::all_variant_names() const
+{
+    if( !can_have_itype_variant() || !has_itype_variant() ) {
+        return std::vector<std::string>( nname( 1 ) );
+    }
+
+    std::vector<std::string> ret;
+    item variant_copy = *this;
+
+    for( const itype_variant_data &iv : type->variants ) {
+        variant_copy.set_variant( iv.id );
+        ret.emplace_back( variant_copy.nname( 1 ) );
+    }
+    return ret;
+}
+
 std::string item::get_corpse_name() const
 {
     return corpse_name;
diff --git a/src/item.h b/src/item.h
index 62bf94e7da..9dfddfa95e 100644
--- a/src/item.h
+++ b/src/item.h
@@ -2500,6 +2500,7 @@ class item : public visitable
          */
         item *get_usable_item( const std::string &use_name );

+        std::vector<std::string> all_variant_names() const;
         /**
          * Returns name of deceased being if it had any or empty string if not
          **/
```